src/olympia/addons/views.py

```diff
@@ -540,6 +540,13 @@ def icloud_bookmarks_redirect(request):
         return addon_detail(request, 'icloud-bookmarks')
 
 
+def find_replacement_addon(request):
+    guid = request.GET.get('guid')
+    if not guid:
+        raise http.Http404
```

Comment:
It wasn't clear what should be displayed if no guid was provided.  I suppose when it's done some generic copy.